pkg/controller/registry/resolver/cache.go

```diff
@@ -45,22 +46,28 @@ type OperatorCacheProvider interface {
 type OperatorCache struct {
 	logger    logrus.FieldLogger
 	rcp       RegistryClientProvider
+	crClient  versioned.Interface
 	snapshots map[registry.CatalogKey]*CatalogSnapshot
 	ttl       time.Duration
 	sem       chan struct{}
 	m         sync.RWMutex
 }
 
+type Priority struct {
+	catsrcPriority int
+}
```

Comment:
```suggestion
type CatalogPriority int32
```

Reply:
dont think this is needed. since the API we merged is int typed

Reply:
It's fine to use int here since its values are only populated from API resources, but my suggestion is primarily to not make this a struct with a single unexported field. If nothing outside of this package will ever use it, then the type itself should be unexported.

Reply:
that i agree.